File: tests/js/openai/test_openllmetry.ts
/**
 * Conformance test: OpenLLMetry (Traceloop) JS OpenAI instrumentation.
 */

import { OpenAIInstrumentation } from "@traceloop/instrumentation-openai";
import { run } from "./common";

function instrument(openaiModule: any) {
  const instrumentation = new OpenAIInstrumentation();
  instrumentation.manuallyInstrument(openaiModule.default);
}

run("OpenLLMetry JS: OpenAI Conformance Test", instrument).catch((e) => { console.error(e); process.exit(1); });

File: tests/python/openai/test_openllmetry.py
"""Conformance test: OpenLLMetry (Traceloop) OpenAI instrumentation."""

from common import run, run_chat, run_chat_streaming, run_chat_tool_call, run_embeddings


def instrument():
    from opentelemetry.instrumentation.openai import OpenAIInstrumentor
    OpenAIInstrumentor().instrument()


if __name__ == "__main__":
    run(
        "OpenLLMetry: OpenAI Conformance Test",
        instrument,
        [run_chat, run_chat_streaming, run_chat_tool_call, run_embeddings],
    )
